views/login/code.pug

```diff
@@ -11,7 +11,9 @@ block content
         div
           label(for='code') Personal access code
           span.pure-form-message For Example: A23XGY12
-          input#code(name='code', type='text', value=data.code, autocomplete='off')
+          input#code(class={['has-error']: errors && errors.code} name='code', type='text', value=data.code, autocomplete='off' aria-describedby=(errors && errors.code ? 'code_error' : false))
```

Comment:
What does this syntax mean? 
ie, `class={['has-error']: errors && errors.code}`?

Reply:
so this is just conditionally checking wether the class should be added, based on if the error for that specific input is present (but we have to check if errors are present at all first).

[pug's got the write up on it here](https://pugjs.org/language/attributes.html) (under class attributes)

Reply:
Ah, I see, the array syntax plus the backwards ternary logic threw me.